internal/config/env.go

```diff
@@ -0,0 +1,50 @@
+package config
+
+import (
+	"log"
+	"strconv"
+
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/gobuffalo/envy"
+)
+
+// ParseEnvironment loads a sibling `.env` file then looks through all environment
+// variables to set global configuration.
+func ParseEnvironment() error {
+	envy.Load()
+	azureEnv, _ := azure.EnvironmentFromName("AzurePublicCloud") // shouldn't fail
+
+	// these must be provided by environment
+	var err1, err2, err3, err4 error
+	clientID, err1 = envy.MustGet("AZURE_CLIENT_ID")
+	clientSecret, err2 = envy.MustGet("AZURE_CLIENT_SECRET")
+	tenantID, err3 = envy.MustGet("AZURE_TENANT_ID")
+	subscriptionID, err4 = envy.MustGet("AZURE_SUBSCRIPTION_ID")
+
+	for _, err := range []error{err1, err2, err3, err4} {
+		if err != nil {
+			return err
+		}
+	}
+
+	// we can choose defaults for these
+	groupName = envy.Get("AZURE_GROUP_NAME", "azure-go-samples")
+	baseGroupName = envy.Get("AZURE_BASE_GROUP_NAME", groupName)
```

Comment:
[nit] I'd make a const for each of the environment variables which can control our test runs. I'd put all of those constants in a single `const` block for the sake of documentation.

Reply:
I think that's what I this is: https://github.com/joshgav/azure-sdk-for-go-samples/blob/33ba00a4453a05b4c9bf84b4f2042a04a1c47708/internal/config/config.go#L14-L30

It's a `var` block cause the values have to be manipulated by `env.go` and `flags.go`.